html/modules/custom/ocha_docstore_files/src/Controller/WebhookController.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+namespace Drupal\ocha_docstore_files\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Logger\LoggerChannelFactoryInterface;
+use Drupal\search_api\Entity\Index;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+
+/**
+ * Controller for incoming webhooks.
+ */
+class WebhookController extends ControllerBase {
+
+  /**
+   * The logger factory.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelFactoryInterface
+   */
+  protected $loggerFactory;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(LoggerChannelFactoryInterface $logger_factory) {
+    $this->loggerFactory = $logger_factory;
+  }
+
+  /**
+   * Download a file.
+   *
+   * @param \Symfony\Component\HttpFoundation\Request $request
+   *   Docstore API request.
+   */
+  public function listen(Request $request) {
+    // Parse JSON.
+    $params = $this->getRequestContent($request);
+
+    if (!isset($params['event'])) {
+      throw new BadRequestHttpException('Illegal payload');
+    }
+
+    $parts = explode(':', $params['event']);
+    $entity_type = $parts[0];
+    $action = $parts[1];
+
+    $uuid = $params['payload']['uuid'];
+    $index = FALSE;
+    $datasource_id = FALSE;
+    $external_entity_type = FALSE;
+
+    // Get index and data source.
+    switch ($entity_type) {
+      case 'knowledge_management':
+        $index = Index::load('km');
+        $datasource_id = 'entity:km';
+        $external_entity_type = 'km';
+        break;
+
+      case 'assessment':
+        $index = Index::load('assessments');
+        $datasource_id = 'entity:assessment';
+        $external_entity_type = 'assessment';
+        break;
+
+      case 'assessment_document':
+        $index = Index::load('assessments');
+        $datasource_id = 'entity:assessment';
+        $external_entity_type = 'assessment_document';
+        break;
+
+      case 'term':
+        $datasource_id = 'organization';
```

Comment:
That looks strange, also no index info.

Reply:
Code needs to be adapted so we can easily handle deletes